service/config_provider.go

```diff
@@ -64,34 +64,41 @@ type ConfigProvider interface {
 }
 
 type configProvider struct {
-	configMapProvider configmapprovider.Provider
-	cfgMapConverters  []ConfigMapConverterFunc
-	configUnmarshaler configunmarshaler.ConfigUnmarshaler
+	configMapProviders []configmapprovider.Provider
+	cfgMapConverters   []ConfigMapConverterFunc
+	configUnmarshaler  configunmarshaler.ConfigUnmarshaler
 
 	sync.Mutex
 	ret     configmapprovider.Retrieved
 	watcher chan error
 }
 
-// newConfigProvider returns a new ConfigProvider that provides the configuration using the given
-// `configMapProvider` and the given `configUnmarshaler`.
-func newConfigProvider(configMapProvider configmapprovider.Provider, configUnmarshaler configunmarshaler.ConfigUnmarshaler, cfgMapConverters ...ConfigMapConverterFunc) ConfigProvider {
+// NewConfigProvider returns a new ConfigProvider that provides the configuration:
+// * Retrieve the config.Map by merging all retrieved maps from all the configmapprovider.Provider in order.
+// * Then applies all the ConfigMapConverterFunc in the given order.
+// * Then unmarshalls the final config.Config using the given configunmarshaler.ConfigUnmarshaler.
+//
+// This is a temporary API.
```

Comment:
Can we add few words what to expect instead going forward?

Reply:
Seconded.  I think I would prefer to see this take functional options as it would allow for more flexibility in how the set of providers and converters are constructed.

Reply:
Even using "options" would not help me, since I will have to do the split of the `ConfigUnmarshaler` which will cause some breaking changes. Until that happens I prefer to mark this experimental to ensure the right expectations.